indico/modules/rb/operations/conflicts.py

```diff
@@ -129,3 +129,16 @@ def get_room_unbookable_hours_conflicts(candidates, occurrences):
                 obj = TempReservationOccurrence(overlap[0], overlap[1], None)
                 conflicts.add(obj)
     return conflicts, conflicting_candidates
+
+
+def get_concurrent_pre_bookings(pre_bookings, skip_conflicts_with=frozenset()):
+    concurrent_pre_bookings = set()
+    for x in pre_bookings:
+        for y in pre_bookings:
```

Comment:
itertools has something better for this (`product` I think); but there might be something even better (`combinations`) sine you don't need `(y, x)` after checking `(x, y)`.